lib/tbot/service_ssh_multiplexer.go

```diff
@@ -435,21 +453,62 @@ func (s *SSHMultiplexerService) Run(ctx context.Context) (err error) {
 
 				var status string
 				switch {
-				case utils.IsOKNetworkError(err):
-					// We reduce the verbosity here since these are to be
-					// expected and are not usually indicative of a problem.
-					status = "OK_NETWORK_ERROR"
-					s.log.DebugContext(egCtx, "Handler exited with a network error", "error", err)
 				case err != nil && !errors.Is(err, context.Canceled):
 					status = "ERROR"
-					s.log.WarnContext(egCtx, "Handler exited with error", "error", err)
+					s.log.WarnContext(egCtx, "Mux handler exited with error", "error", err)
 				default:
 					status = "OK"
 				}
 				muxReqsHandledCounter.WithLabelValues(status).Inc()
 			}()
 		}
 	})
+	// Handle agent listener
+	eg.Go(func() error {
+		defer context.AfterFunc(egCtx, func() { _ = agentListener.Close() })()
+		for {
+			conn, err := agentListener.Accept()
+			if err != nil {
+				if utils.IsUseOfClosedNetworkError(err) {
+					return nil
+				}
+
+				s.log.WarnContext(
+					egCtx,
+					"Error encountered accepting agent connection, sleeping and continuing",
+					"error",
+					err,
+				)
+				select {
+				case <-egCtx.Done():
+					return nil
+				case <-time.After(50 * time.Millisecond):
+				}
+
+				continue
+			}
+
+			go func() {
+				defer context.AfterFunc(egCtx, func() { _ = conn.Close() })()
```

Comment:
Why do we need to defer these `AfterFunc` calls?

Reply:
It removes the watcher on the `context.Context` e.g if the root context isn't cancelled and proxying exits for another reason, we don't want to continue waiting for the context to be cancelled.